exporter/datadogexporter/internal/logs/translator.go

```diff
@@ -0,0 +1,180 @@
+
+	l := datadogV2.HTTPLogItem{
+		AdditionalProperties: make(map[string]string),
+	}
+	if hostName != "" {
+		l.Hostname = datadog.PtrString(hostName)
+	}
+
+	if serviceName != "" {
+		l.Service = datadog.PtrString(serviceName)
+	}
+
+	// we need to set log attributes as AdditionalProperties
+	lr.Attributes().Range(func(k string, v pcommon.Value) bool {
+		l.AdditionalProperties[k] = v.AsString()
+		return true
+	})
+	if !lr.TraceID().IsEmpty() {
+		l.AdditionalProperties[ddTraceID] = strconv.FormatUint(traceIDToUint64(lr.TraceID()), 10)
+		l.AdditionalProperties[otelTraceID] = lr.TraceID().HexString()
+	}
+	if !lr.SpanID().IsEmpty() {
+		l.AdditionalProperties[ddSpanID] = strconv.FormatUint(spanIDToUint64(lr.SpanID()), 10)
+		l.AdditionalProperties[otelSpanID] = lr.SpanID().HexString()
+	}
+	var status string
+
+	// we want to use the serverity that client has set on the log and let datadog backend
+	// decide the appropriate level
+	if lr.SeverityText() != "" {
+		status = lr.SeverityText()
+		l.AdditionalProperties[otelSeverityText] = lr.SeverityText()
+	} else if lr.SeverityNumber() != 0 {
+		status = derviveStatusFromSeverityNumber(lr.SeverityNumber())
+	}
+
+	l.AdditionalProperties[ddStatus] = status
+	// if SeverityNumber is set , we want to retain it
+	if lr.SeverityNumber() != 0 {
+		l.AdditionalProperties[otelSeverityNumber] = strconv.Itoa(int(lr.SeverityNumber()))
+	}
+
+	// for datadog to use the same timestamp we need to set the additional property of "@timestamp"
+	if lr.Timestamp() != 0 {
+		// we are retaining the nano second precision in this property
+		l.AdditionalProperties[otelTimestamp] = strconv.FormatInt(lr.Timestamp().AsTime().UnixNano(), 10)
+		l.AdditionalProperties[ddTimestamp] = lr.Timestamp().AsTime().Format(time.RFC3339)
+	}
+
+	if sendLogRecordBody {
+		// set the Message
+		l.Message = lr.Body().AsString()
+	}
+
+	var tags = attributes.TagsFromAttributes(res.Attributes())
+	if len(tags) > 0 {
+		tagStr := strings.Join(tags, ",")
+		l.Ddtags = datadog.PtrString(tagStr)
+	}
+	return l
+}
+
+func extractHostNameAndServiceName(resourceAttrs pcommon.Map, logAttrs pcommon.Map) (hostName string, serviceName string) {
+	if src, ok := attributes.SourceFromAttributes(resourceAttrs, true); ok && src.Kind == source.HostnameKind {
+		hostName = src.Identifier
+	}
+
+	// hostName is blank from resource
+	// we need to derive from log attributes
+	if hostName == "" {
+		if src, ok := attributes.SourceFromAttributes(logAttrs, true); ok && src.Kind == source.HostnameKind {
+			hostName = src.Identifier
+		}
+	}
+
+	if s, ok := resourceAttrs.Get(conventions.AttributeServiceName); ok {
+		serviceName = s.AsString()
+	}
+
+	// serviceName is blank from resource
+	// we need to derive from log attributes
+	if serviceName == "" {
+		if s, ok := logAttrs.Get(conventions.AttributeServiceName); ok {
+			serviceName = s.AsString()
+		}
+	}
+
+	return hostName, serviceName
```

Comment:
```suggestion
	if src, ok := attributes.SourceFromAttributes(resourceAttrs, true); ok && src.Kind == source.HostnameKind {
		hostName = src.Identifier
	}
	// hostName is blank from resource
	// we need to derive from log attributes
	if hostName == "" {
		if src, ok := attributes.SourceFromAttributes(logAttrs, true); ok && src.Kind == source.HostnameKind {
			hostName = src.Identifier
		}
	}
	if s, ok := resourceAttrs.Get(conventions.AttributeServiceName); ok {
		serviceName = s.AsString()
	}
	// serviceName is blank from resource
	// we need to derive from log attributes
	if serviceName == "" {
		if s, ok := logAttrs.Get(conventions.AttributeServiceName); ok {
			serviceName = s.AsString()
		}
	}
	return hostName, serviceName
```
Could we keep it more compact?